### ISSUE TITLE
[bug]: MM2 downloads extra files from diffusers repos

### ISSUE THREAD

**Issue by psychedelicious** [invoke-ai/InvokeAI#5861]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### Operating system

Linux

### Version number

https://github.com/invoke-ai/InvokeAI/pull/5846

### Browser

FF

### What happened

I think MM2 maybe be installing the wrong set of files for HF diffusers repos. I noticed that I was downloading an extra 1GB when testing the installer with sd1.5.

Here are the files selected for `runwayml/stable-diffusion-v1-5`:

```py
[
    ".../feature_extractor/preprocessor_config.json"
    ".../model_index.json"
    ".../safety_checker/config.json"
    ".../safety_checker/model.fp16.safetensors"  # Extraneous
    ".../safety_checker/pytorch_model.fp16.bin"
    ".../scheduler/scheduler_config.json"
    ".../text_encoder/config.json"
    ".../text_encoder/model.fp16.safetensors"  # Extraneous
    ".../text_encoder/pytorch_model.fp16.bin"
    ".../tokenizer/merges.txt"
    ".../tokenizer/special_tokens_map.json"
    ".../tokenizer/tokenizer_config.json"
    ".../tokenizer/vocab.json"
    ".../unet/config.json"
    ".../unet/diffusion_pytorch_model.fp16.safetensors"  # Shouldn't we download the .bin?
    ".../vae/config.json"
    ".../vae/diffusion_pytorch_model.fp16.safetensors"  # Shouldn't we download the .bin?
]
```

### What you expected to happen

Correct files are downloaded

### How to reproduce the problem

Use MM2 on `main` and install `runwayml/stable-diffusion-v1-5`. Expect to download ~3.3GB. The download should be more like ~2.3GB.

**Comment by psychedelicious** [comment]
On closer inspection I see we are filtering the files in `invokeai/backend/model_manager/util/select_hf_files.py`, and there is a note about how we prefer `.safetensors` files, so never mind mine comment about downloading `.bin` files.

The remaining issue downloading extra safetensors files. This occurs because the `safetensors` and `bin` files have different names, despite representing the same weights. 

For example, `text_encoder/model.fp16.safetensors" and `text_encoder/pytorch_model.fp16.bin` are the same, but with different base names - "model" vs "pytorch_model". I don't see any simple way to fix.

**Comment by psychedelicious** [comment]
Figured out a fix in #5846.

We only want the single "best" weights file for each submodel type/directory. We prefer safetensors and exact variant matches. To choose the right one, we can score each file and take the best one.
